src/Planner/PlannerJoinTree.cpp

```diff
@@ -1102,6 +1139,17 @@ JoinTreeQueryPlan buildQueryPlanForTableExpression(QueryTreeNodePtr table_expres
                 }
 
                 auto & alias_column_expressions = table_expression_data.getAliasColumnExpressions();
+                if (has_table_virtual_column && query_plan.isInitialized())
+                {
+                    const auto & data_header = query_plan.getCurrentHeader();
+                    if (!data_header.findByName("_table"))
+                    {
+                        const auto & table_name = storage->getStorageID().getTableName();
+                        auto adding_table_name_virtual_column_step = std::make_unique<AddingTableNameVirtualColumnStep>(data_header, table_name);
```

Comment:
Please do not create a new step for this.
Use ExpressionStep (create and ActionsDAG which adds a constant string column, it should be trivial)

Reply:
@KochetovNicolai Used ExpressionStep as an alternative.